Adding payload samples

rbills opened the pull request:
I've added in payload sample templates for Tasks (templates and tasks), change requests, and modified the shared API folder.
I've not made any changes to the ReadMe or walk through files.

Please review for acceptance and help make sure I haven't accidentally changed the markdown files.

rbills committed:
Updating names for new APT-SCWM terminology.

rbills committed:
adding new task payload template files

rbills committed:
rearranging the Shared API folder

rbills committed:
updating file and folder names

rbills committed:
adding new task template payload samples

rbills committed:
adding change request payload templates

knmeiss commented:
@rbills I don't think security is going to be ok with having any valid ID's shown throughout the repository (e.g ownerId, processNetworkId, currentlyAssignedPartnerUsersId, etc.) How would you feel about redacting those values? I'd also be happy to reach out to my contact on the security team to review/confirm.

rbills committed:
adding payload header info to mirror the API docs

rbills committed:
restoring payload header info to indirect supplier incident samples

rbills committed:
restoring payload header info to external manufacturer incident samples

rbills committed:
restoring payload header info to internal manufacturer incident samples

rbills committed:
restoring payload header info to shared incident API samples

rbills committed:
substituting fictitious identifiers and latin phrases for placeholder values

rbills committed:
updating Python instructions to remove 'wrapper-style' instructions

rbills committed:
removing >>> prompts from document for clarity

rbills committed:
editing the coe sample to show how to load JSON from a file

rbills committed:
interim commit while updating readme docs

rbills committed:
updating to remove references to utils.py methods

rbills committed:
fixing typos

rbills committed:
fixing broken link

rbills committed:
updating walkthrough according to Python test run

rbills committed:
adding some extra help to the walkthrough

rbills committed:
adding a couple of local variable declarations for clarity

rbills committed:
fixing a typo

rbills committed:
standardizing formatting

rbills committed:
fixing a format error for a variable

rbills committed:
revising for readabilitu

rbills committed:
revising for consistency of variable names

rbills committed:
rewording for clarity

rbills committed:
fixing misplaced punctuation

rbills commented:
I changed anything that looked like an ID to a value placeholder.  I have faith our developer customers will understand what's expected.
I also modified the QuickStart.MD to provide a fully 'Pythonic' method of making calls to the API.  We create a payload in Python, modify one of the values, cast it to JSON, make a request and then re-use the generated ID to add a comment and then view the incident.

rbills committed:
adding clarification for the json.dumps command

knmeiss commented on python/Quickstart.MD:
````diff
@@ -5,130 +5,107 @@ This quickstart covers:
 ```
-
-However, for this example, we will include more than the miniumum information. Create a payload file containing the below and name it `addIncident.json`.  
+**Type or copy the following into your Python terminal**  
````
**Type or copy the following into your Python terminal, replacing necessary values**

^^ do this with all that read this way

rbills replied:
Updated as suggested.

knmeiss commented on python/Quickstart.MD:
````diff
@@ -5,130 +5,107 @@ This quickstart covers:
+  }
 }
 ```
+This creates a dictionary object containing the `header` and `payload` elements we discussed earlier.  You can work with the items in this dictionary using standard Python methods.  This allows you to make any desired modifications prior to packaging and submitting your request.  
````
- Eliminate word "we" in first sentence. 
- I think you can remove the second two sentences, this is assumed for developers. Suggested edit: 
"This creates a dictionary object containing the `header` and `payload` elements discussed earlier. Make any necessary changes before submitting the request. For example..."

rbills replied:
Updated as suggested

knmeiss commented on python/Quickstart.MD:
```diff
@@ -5,130 +5,107 @@ This quickstart covers:
+This creates a dictionary object containing the `header` and `payload` elements we discussed earlier.  You can work with the items in this dictionary using standard Python methods.  This allows you to make any desired modifications prior to packaging and submitting your request.  
 
-The `utils.py` file has functions to help build the payload for your request body. The `read_payload_file(filename)` function reads the contents of the file, loads it into a JSON object, and does some minor processing for insertion into the request body.
+As an example, let's change the priority of this incident from `HIGH` to `MEDIUM`.  To change this value, we must reference the `businessPriority` item.  To reach this value, reference the `directSupplierImpact` element of the `payload`.  
```
eliminate "let's" and "we" 

For example, change the priority of this incident from `HIGH` to `MEDIUM` by referencing the `businessPriority` item. To reach this value, reference the `directSupplierImpact` element of the `payload`.

rbills replied:
Updated as suggested

knmeiss commented on python/Quickstart.MD:
````diff
@@ -5,130 +5,107 @@ This quickstart covers:
+request_body.get('payload').get('directSupplierImpact').update({'businessPriority': 'MEDIUM'})
 ```
-If you already use a JSON payload as part of your other processes, you can the `accept_json_payload(payload_in)` function instead. This function accepts a JSON object, checks for proper format, and does some minor processing for insertion into the request body.
+We use the `.get()` method for each element in our tree that we wish to access.  The `.update()` method allows us to change a value by referencing its key.
````
eliminate we and us

"Use the `.get()` method for each element you wish to access. The `.update()` method allows for changing a value by referencing its key."

rbills replied:
Reworded

knmeiss commented on python/Quickstart.MD:
````diff
@@ -5,130 +5,107 @@ This quickstart covers:
-```
->>> my_json_data = result_from_some_other_process()
->>> payload_data = utils.accept_json_payload(my_json_data)
+Take a moment to add your `ownerId` and `processNetworkId` values using the method outlined above.  Some sample code is below as a hint:  
````
I think this should be the first step -- move up and include with my earlier feedback about wording

Eliminate/reword "Some sample code is below as a hint:" Suggestion: 

"You should now be have a request body similar to the sample below"

rbills replied:
I want to leave this here.  If we establish the variables before the payload, the only advantage we get is that we can build the whole request dynamically.  What we lose is the opportunity to introduce the update method before we get too deep into the instruction.  This update method will be very important for most of the automation process.

knmeiss commented on python/Quickstart.MD:
```diff
@@ -5,130 +5,107 @@ This quickstart covers:
 
-Now that you have usable payload data, you need to inform APT how to process the payload.  
+### Request Header
+APT-SCWM uses tokens for authentication.  For single interactive requests, you can use a bearer token (a token that usually expires within an hour) or a longer-lived token for automated processes.  The token is a string that contains 'user:password' in Base64 encoding.
```
Reword to eliminate ( ) ... suggestion below 

"APT-SCWM uses tokens for authentication. For single interactive requests, use a short-lived bearer token, which lasts for an hour, or a longer-lived token for automated processes." 

You could link back out to auth documentation in repo

rbills replied:
Updated.  Not linking at this moment.

knmeiss commented on python/Quickstart.MD:
````diff
@@ -5,130 +5,107 @@ This quickstart covers:
+request_headers = {"Authorization": "{0} {1}".format(method, token), "Content-Type": "application/json"}
 ```
-This dictionary is passed in as a parameter to the `utils.create_payload(event_in, payload_in)` function.  This function places the header and payload JSON items properly in the request body.  
+The resulting `headers` variable will look similar to  
````
add : after to

rbills replied:
Updated

knmeiss commented on python/Quickstart.MD:
````diff
@@ -5,130 +5,107 @@ This quickstart covers:
->>> request_body = utils.create_payload(event_data, payload_data)
-```
+## Package Your Request  
+At this point you have a request body and a request header.  You have made all required modifications using the Python dictionary methods.  APT-SCWM requires requests formatted as JSON (JavaScript Object Notation).  
````
"At this point you should have a request body, request header, and made all required modifications using Python dictionary methods."

rbills replied:
Updated

knmeiss commented on python/Quickstart.MD:
```diff
@@ -5,130 +5,107 @@ This quickstart covers:
+Until now, we have been using a Python terminal without any additional packages.  It's time to add the JSON library so we can translate our dictionary into a format accepted by APT-SCWM.  The JSON library is included in the Python installation by default, so no extra installations are required.  
 
-The `utils.py` file provides a function to create a request header that can be reused until token expiration.  
+You may wish to verify your request prior to submission.  The `json.dumps()` method of the json library can help.  Use this method against your existing `request_body` dictionary to see what the `json.dumps()` method will generate.  
```
If you wish to verify your request prior to submission, use the `json.dumps()` method of the json library against your existing `request_body` dictionary.

rbills replied:
Updated

knmeiss commented on python/Quickstart.MD:
````diff
@@ -142,192 +119,201 @@ Body of event request:
+    }
 }
+```  
+Note that your `False` values have been converted to `false` according to the JSON specification.  Everything looks good, so we can convert our dictionary to a JSON string.  
````
Note that your `False` values have been converted to `false` according to the JSON specification. Now, can convert the dictionary to a JSON string.

rbills replied:
Reworded

knmeiss commented on python/Quickstart.MD:
````diff
@@ -142,192 +119,201 @@ Body of event request:
+```
+request_body_json = json.dumps(request_body)
 ```
+Note that we have omitted the `indent=4, sort_keys=False` options from the `json.dumps()` command that we included in the print statement.  We used these for clarity when print, but are not necessary when creating our JSON string.
````
I think this can be removed? 
This should be obvious to a developer

rbills replied:
It should be obvious to experienced developers, but those new to the language or Python's JSON library can miss it.  I want to accommodate people thrown into the automation role as well as experienced developers.

knmeiss commented on python/Quickstart.MD:
```diff
@@ -142,192 +119,201 @@ Body of event request:
+Note that we have omitted the `indent=4, sort_keys=False` options from the `json.dumps()` command that we included in the print statement.  We used these for clarity when print, but are not necessary when creating our JSON string.
 
 ## Send Request
+Once your request header and request body are complete you are ready to submit your request.  
```
Once the request header and request body are complete the request is ready to send.

rbills replied:
Updated.  Kept the word 'submit' for consistency.

knmeiss commented on python/Quickstart.MD:
```diff
@@ -142,192 +119,201 @@ Body of event request:
 
 ## Send Request
+Once your request header and request body are complete you are ready to submit your request.  
+You will need the URL for the APT-SCWM environment.  Contact your administrator for this information.  
```
This requires a URL for the APT-SCWM environment. Contact your administrator for this information.

rbills replied:
updated

knmeiss commented on python/Quickstart.MD:
```diff
@@ -142,192 +119,201 @@ Body of event request:
 ## Send Request
+Once your request header and request body are complete you are ready to submit your request.  
+You will need the URL for the APT-SCWM environment.  Contact your administrator for this information.  
+APT-SCWM uses and event driven API that communicates using the POST verb of the request/response model.  Create a variable to hold the response so you can work with your results.  
```
the word and should be "an"

Can this also be reworded to: 

APT-SCWM uses an event driven API that communicates using the POST verb for all API calls.

I think the rest can be eliminated because that should be obvious to developers.

rbills replied:
Fixed

knmeiss commented on python/Quickstart.MD:
````diff
@@ -142,192 +119,201 @@ Body of event request:
->>> response_json = json.loads(result.text)
->>> incident_id = response_json.get('payload').get('id')
-```
+The `id` in a successful response payload is the identifier of the created incident. Notice that the response is a JSON object that has been converted to a string for transmission. To work with this data, convert it to a Python dictionary.  
````
Notice that the response is a JSON object that is converted to a string for transmission.

rbills replied:
Fixed

knmeiss commented on python/Quickstart.MD:
````diff
@@ -142,192 +119,201 @@ Body of event request:
+incident_id = response_json.get('payload').get('id')
 ```
 
+You may now reference your created incident id by using its variable name.
````
eliminate this

rbills replied:
removed

knmeiss commented on python/Quickstart.MD:
```diff
@@ -142,192 +119,201 @@ Body of event request:
 
-Prepare a comment payload file called `comment_payload.json` containing the below:
+The APT-SCWM shared API contains processes for adding, editing, or listing comments for incidents along with other utility functions.  The header of this request body is slightly different (in this case only the `eventName` changes) and uses the same authorization configuration as the previous request.  
+If you are building or modifying your requests programmatically, you could use the `ITEM_ID` you created in the previous step.  The JSON library allows us to use `json.loads(string)` to create a dictionary from an existing JSON object and then use the `json.dumps(dictionary)` to convert our modified dictionary back to a JSON object.  
```
"If you are building or modifying your requests programmatically, use the `ITEM_ID` created in the previous step. The JSON library allows using `json.loads(string)` to create a dictionary from an existing JSON object, followed by `json.dumps(dictionary)` to convert the modified dictionary back to a JSON object."

rbills replied:
reworded

knmeiss commented on python/Quickstart.MD:
````diff
@@ -142,192 +119,201 @@ Body of event request:
+  }
 }
 ```
+Remember that we created a dictionary object with the above data.  If we wish to update the `ITEM_ID` with the result from the previous step, we can use the `.update()` method.  
````
Eliminate first sentence 

"To update the result, use the `ITEM_ID` and the `.update()` method."

rbills replied:
reworded

knmeiss commented on python/Quickstart.MD:
````diff
@@ -142,192 +119,201 @@ Body of event request:
 }
 ```
+Remember that we created a dictionary object with the above data.  If we wish to update the `ITEM_ID` with the result from the previous step, we can use the `.update()` method.  
+Let's make our modifications and add a comment.  
````
eliminate

rbills replied:
removed

knmeiss commented on python/Quickstart.MD:
````diff
@@ -142,192 +119,201 @@ Body of event request:
+comment_request.get('payload').update({'processId': incident_id})
 ```
-The payload and event data should now be updated. You can verify this by resending the commands to build the request body and resend the request.  
+After making our desired modifications, we can convert our dictionary to a JSON string and send our request.  
````
"After making the desired modifications, convert the dictionary to a JSON string and send the request."

rbills replied:
reworded

knmeiss commented on python/Quickstart.MD:
````diff
@@ -142,192 +119,201 @@ Body of event request:
 }
 ```
 ## Read the details of your newly created incident
 In a previous step, you stored your incident id as a variable in your session.  You can use this stored variable to retrieve details about the incident.  
````
Use the previously stored incident id to retrieve details about the incident.

rbills replied:
updated

knmeiss commented on python/Quickstart.MD:
````diff
@@ -142,192 +119,201 @@ Body of event request:
    }
 }
 ```
+You can then prepare and send your request.  
````
Prepare and send your request.

rbills replied:
Updated

knmeiss commented on python/Quickstart.MD:
````diff
@@ -341,22 +327,23 @@ The response provides the complete record for the incident:
 }
 }
 ```
+Remember, you can use the `json.loads(string)` method to convert this result to a useful dictionary object for further processing.  
````
eliminate

rbills replied:
removed

rbills committed:
fixing the payload body

jcleblanc commented on python/payload_samples/ChgRequest/addChangeRequestAllFieldsPayloadTemplate.json:
```diff
@@ -0,0 +1,94 @@
+      "partnerId": "PARTNER_USER_ID",
+      "toIdType": "companyMasterData"
+    },
+    "subjectMatterExpert": [
```
For any direct contact information used in these samples like these it may be valuable to use standard persona profiles from tech pubs (reach out to them for details). Until that's ready (to not bottleneck this release) I'd recommend removing this direct contact information for BSpitzer@gmail.com and the name with something generic, as it looks like BSpitzer@gmail.com is associated with a real person.

rbills replied:
OK, those are copied from the API docs page.  Do we pull that out for now, or put something more generic in?

rbills replied:
changed to something obviously fictitous

jcleblanc replied:
👍🏻 thanks for the adjustment. Once we have these in place and techpubs revisits adding guides we can provide them with these adjusted values.

jcleblanc commented on python/payload_samples/ChgRequest/addChangeRequestAllFieldsPayloadTemplate.json:
```diff
@@ -0,0 +1,94 @@
+        },
+        "addAttachment": [
+          {
+            "fileName": "82kb",
```
I think the values for file name and file size here are swapped

rbills replied:
I'll fix that.  That's actually how it looks in the docs. lol

rbills replied:
fixed

jcleblanc commented on python/payload_samples/ChgRequest/copyChangeRequestPayloadTemplate.json:
```diff
@@ -0,0 +1,20 @@
+    "dataspace": "default"
+  },
+  "payload": {
+    "id": "YOUR_PROCESS_NETWORK_ID3",
```
Remove the 3 from YOUR_PROCESS_NETWORK_ID3

rbills replied:
fixed

jcleblanc commented on python/payload_samples/ChgRequest/editChangeRequestPartnerImpactPayloadTemplate.json:
```diff
@@ -0,0 +1,56 @@
+      "toId": "COMPANY_ID",
+      "toIdType": "companyMasterData"
+    },
+    "subjectMatterExpert": [
```
As previously commented, let's use standard generic profiles from techpubs here for the long term and generic data (e.g. John / Jane Smith) in the interim until that's available.

rbills replied:
fixed

jcleblanc commented on python/payload_samples/ChgRequest/editChangeRequestPartnerImpactPayloadTemplate.json:
```diff
@@ -0,0 +1,56 @@
+    "impactedLocation": [
+      {
+        "locationId": "LOCATION_ID",
+        "locationType": "companyLocationMasterData",
```
I believe locationType should be INTERNAL or EXTERNAL according to the docs

rbills replied:
fixed

jcleblanc commented on python/payload_samples/DirSupIncident/addDirSupIncidentPayloadTemplate.json:
```diff
@@ -1,154 +1,162 @@
-        "partnerCompany": {
-            "partnerId": "be9c8cea-72bc-459c-97b9-ef3f7c73ccc8",
-            "toIdType": "companyMasterData"
+PARTNER_USER_ID{
```
Remove PARTNER_USER_ID (I believe it was accidentally included)

jcleblanc commented on python/payload_samples/ExtMfgIncident/editMfgIncPartnerImpactPayloadTemplate.json:
```diff
@@ -1,62 +1,70 @@
-                "visibilityType": "Internal"
-            }
-        }
+PRODUCT_ID{
```
Remove PRODUCT_ID

jcleblanc commented on python/payload_samples/IndirSupIncident/addIndSupIncPayloadTemplate.json:
```diff
@@ -1,141 +1,148 @@
+    "currentlyAssignedCompanyUsers": [
+      "COMPANY_USER_ID"
+    ],
+    "currentlyAssignedPartnerUsers": [
```
It might be good to differentiate that these are two separate partners by using PARTNER_USER_ID1 and PARTNER_USER_ID2

jcleblanc commented on python/payload_samples/IndirSupIncident/addIndSupIncPayloadTemplate.json:
```diff
@@ -1,141 +1,148 @@
+      "highPriorityReasons": [
+        "Reason1"
+      ],
+      "financialImpact": 164113394500,
```
While this value (financialImpact) would be fine to pass through, I think this is a copied timestamp. According to the docs this field should be a number and should include, "The monetary impact that the incident has on the Owner, entered as a numeric with decimals (e.g. 150000.00). Cannot exceed 255 characters."

rbills replied:
fixed

jcleblanc commented on python/payload_samples/IndirSupIncident/editIndSupIncPartnerPayloadTemplate.json:
```diff
@@ -1,56 +1,64 @@
+    },
+    "partnerLocationId": "LOCATION_ID",
+    "indirectPartnerLocationId": "LOCATION_ID",
+    "responsiblePartyAtPartner": "COMMENT_ID",
```
responsiblePartyAtPartner should be the string identifier for the user, or null for unassigned

rbills replied:
fixed

jcleblanc commented on python/payload_samples/IntMfgIncident/AddIntMfgIncFullPayloadTemplate.json:
```diff
@@ -1,98 +1,106 @@
+    "aptBusinessObjectSummary": "Missing Shipment",
+    "aptBusinessObjectDescription": "Shipment XYZ missing at Location ABC",
+    "deviationType": "PLANNED",
+    "responsiblePartyAtCompany": "COMPANY_USER_ID",
```
Should responsiblePartyAtCompany be a user ID or is it another identifier type? I've seen edavis used previously in fields like these. Just need to confirm this.

rbills replied:
The docs say responsiblePartyAtCompany is an identifier.  The sample in the doc is a guid.

jcleblanc commented on python/payload_samples/IntMfgIncident/AddIntMfgIncFullPayloadTemplate.json:
```diff
@@ -1,98 +1,106 @@
+      "initialContainmentActions": "Final Containment Actions",
+      "immediateCorrectiveActions": "Immediate Corrective Actions",
+      "potentialRootCauses": [
+        "DOCUMENTATION"
```
More a comment, it looks like most of the samples I've seen thus far with root causes all use DOCUMENTATION, we might want to switch this up with the other cause types to round out the sample data a bit.

rbills replied:
Added a generic statement

jcleblanc commented on python/payload_samples/IntMfgIncident/EditIntMfgIncImpactPayloadTemplate.json:
```diff
@@ -1,37 +1,45 @@
+      "highPriorityReasons": [
+        "Reason1"
+      ],
+      "financialImpact": 164113394500,
```
financialImpact should be the numeric value representing the monetary value - I think this is a timestamp.

rbills replied:
fixed

jcleblanc commented on python/payload_samples/Shared/getActivityHistoryPayloadTemplate.json:
```diff
@@ -0,0 +1,14 @@
+  },
+  "payload": {
+    "id": "ITEM_ID",
+    "processType": "incident"
```
Can you verify that processType should be incident here please? I see conflicting information in the docs, the table says it should be more specific (e.g. directSupplierIncident) but the payload sample included shows incident. This may be fine, but just need to verify.

rbills replied:
incident is a valid value when you reference a generic incident.  I have used this specific value successfully.

jcleblanc commented on python/payload_samples/Task/addTaskSubTaskPayloadTemplate.json:
```diff
@@ -0,0 +1,58 @@
+      "COMPANY_USER_ID"
+    ],
+    "currentlyAssignedPartnerUsers": [
+      "PARTNER_USER_ID",
```
Recommend differentiating two separate partners here by using PARTNER_USER_ID1 and PARTNER_USER_ID2

rbills replied:
fixed

jcleblanc commented on python/payload_samples/Task/closeSubTaskPayloadTemplate.json:
```diff
@@ -0,0 +1,13 @@
+    "dataspace": "default"
+  },
+  "payload": {
+    "id": "YOUR_PROCESS_NETWORK_ID3"
```
Remove 3 from YOUR_PROCESS_NETWORK_ID3

rbills replied:
fixed

jcleblanc commented on python/payload_samples/Task/editSubTaskPayloadTemplate.json:
```diff
@@ -0,0 +1,18 @@
+  },
+  "payload": {
+    "id": "ITEM_ID",
+    "currentlyAssignedPartnerUsers": [
```
Recommend PARTNER_USER_ID1 & 2 to differentiate if they should be different partners

rbills replied:
fixed

jcleblanc commented on python/payload_samples/Task/editTaskPayloadTemplate.json:
```diff
@@ -0,0 +1,14 @@
+TASK_ID{
```
Remove TASK_ID

jcleblanc commented on python/payload_samples/Task/listSubTaskForTaskPayloadTemplate.json:
```diff
@@ -0,0 +1,13 @@
+    "dataspace": "default"
+  },
+  "payload": {
+    "id": "YOUR_PROCESS_NETWORK_ID3"
```
Remove 3 from YOUR_PROCESS_NETWORK_ID3

rbills replied:
fixed

jcleblanc commented on python/payload_samples/Task/removeSubTaskFromTaskPayloadTemplate.json:
```diff
@@ -0,0 +1,13 @@
+    "dataspace": "default"
+  },
+  "payload": {
+    "id": "YOUR_PROCESS_NETWORK_ID3"
```
Remove 3 from YOUR_PROCESS_NETWORK_ID3

rbills replied:
fixed

jcleblanc commented on python/payload_samples/Task/reopenSubTaskPayloadTemplate.json:
```diff
@@ -0,0 +1,13 @@
+    "dataspace": "default"
+  },
+  "payload": {
+    "id": "YOUR_PROCESS_NETWORK_ID3"
```
Remove 3 from YOUR_PROCESS_NETWORK_ID3

rbills replied:
fixed

jcleblanc commented on python/payload_samples/TaskTemplate/addSubTaskToTemplatePayloadTemplate.json:
```diff
@@ -0,0 +1,43 @@
+    },
+    "responsiblePartyAtPartner": "PARTNER_USER_ID",
+    "responsibleDepartmentAtPartner": "SUPPLY_CHAIN",
+    "currentlyAssignedCompanyUsers": [
```
Recommend differentiating with 1 & 2

rbills replied:
fixed

jcleblanc commented on python/payload_samples/TaskTemplate/addTaskTemplatePayloadTemplate.json:
```diff
@@ -0,0 +1,21 @@
+    "businessPriority": "LOW",
+    "responsibleDepartmentAtCompany": "CUSTOMER_SERVICE",
+    "responsiblePartyAtCompany": "COMPANY_USER_ID",
+    "currentlyAssignedCompanyUsers": []
```
Recommend adding a value in here for the docs - it's valid as null but would be valuable to showcase something for the value.

rbills replied:
added a placeholder

jcleblanc commented on python/payload_samples/TaskTemplate/copyTaskTemplatePayloadTemplate.json:
```diff
@@ -0,0 +1,16 @@
+    "dataspace": "default"
+  },
+  "payload": {
+    "id": "COMMENT_ID",
```
I think this should be TASK_TEMPLATE_ID

rbills replied:
fixed

jcleblanc commented on python/payload_samples/TaskTemplate/editSubTaskOnTemplatePayloadTemplate.json:
```diff
@@ -0,0 +1,23 @@
+    "dataspace": "default"
+  },
+  "payload": {
+    "id": "ITEM_ID",
```
Should be TEMPLATE_ID

rbills replied:
fixed

jcleblanc commented on python/payload_samples/TaskTemplate/editSubTaskOnTemplatePayloadTemplate.json:
```diff
@@ -0,0 +1,23 @@
+      "COMPANY_USER_ID"
+    ],
+    "currentlyAssignedPartnerUsers": [
+      "PARTNER_USER_ID",
```
Differentiate with 1 & 2

rbills replied:
fixed

jcleblanc commented on python/payload_samples/TaskTemplate/editTaskTemplatePayloadTemplate.json:
```diff
@@ -0,0 +1,19 @@
+    "dataspace": "default"
+  },
+  "payload": {
+    "id": "COMMENT_ID",
```
Should be TEMPLATE_ID

rbills replied:
fixed

jcleblanc commented on python/payload_samples/TaskTemplate/listSubTaskForTemplatePayloadTemplate.json:
```diff
@@ -0,0 +1,13 @@
+    "dataspace": "default"
+  },
+  "payload": {
+    "id": "ITEM_ID"
```
Should be TEMPLATE_ID

rbills replied:
fixed

jcleblanc commented on python/payload_samples/TaskTemplate/removeSubTaskfromTemplatePayloadTemplate.json:
```diff
@@ -0,0 +1,13 @@
+    "dataspace": "default"
+  },
+  "payload": {
+    "id": "ITEM_ID"
```
Should be SUB_TASK_ID (or SUBTASK_ID)

rbills replied:
fixed

rbills committed:
fixing items found in merge review

rbills committed:
removing unwanted file

jcleblanc reviewed (changes requested):
Code samples / guides are looking good - a few items in general that I added down:

- Payload updates - I added a number of comments for specific payload updates
- .DS_Store file under the payload_samples folder - please delete this from the PR as it's a local macOS file that's generated.
- Standard persona information - As a followup item when using PII like email / name we should probably have standard personas from techpubs. In one instance the email was associated with a real person, but I believe those come straight from the docs. For the time being you can make these generic (John / Jane Smith) and then we can follow up with techpubs for longer term requirements.

rbills committed:
fixing items found during review

jcleblanc reviewed (approved):
Changes look good from my side, thanks Rob

rbills committed:
fixing a missed edit
